js/tests/unit/scrollspy.js

```diff
@@ -231,48 +235,40 @@ $(function () {
 
   QUnit.test('should correctly select navigation element on backward scrolling when each target section height is 100%', function (assert) {
     assert.expect(5)
+    var done = assert.async()
+
+    var navLinks = ''
+    var contentDivs = ''
+    var i = 0
+
+    while (++i < 6) {
+      navLinks += '<li id="li-100-' + i + '"><a href="#div-100-' + i + '">div ' + i + '</a></li>'
+      contentDivs += '<div id="div-100-' + i + '" style="position: relative; height: 100%; padding: 0; margin: 0">div ' + i + '</div>'
+    }
+
     var navbarHtml =
         '<nav class="navbar">'
       + '<ul class="nav">'
-      + '<li id="li-100-1"><a href="#div-100-1">div 1</a></li>'
```

Comment:
No i understand, i was the same way for a while – but in reality, bootstrap's js is already pretty tiny (particularly after uglify gets at it) – if anything, i think our currently library code suffers from being too clever – which just makes things harder for others to follow, without actually saving ~that~ many bytes.

I think in v4 we will be making some efforts to unpack some of the "dryness" in an effort to improve readability.

It's really just a different approach – i suppose we could document that somewhere though.